### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by msssk**
* update from es5 to es2015 (remove bluebird)
* migrate from tsd typings to @types
* update Intern from 3 to 4
* remove obsolete TSLint rules
* fix lint/syntax issues

I updated the version in package.json to 3.0.0-alpha since this is a breaking change - the compile target is now es2015, so it must be run in an environment that supports es2015 (or be transpiled and shimmed). Intern 4 appears to depend on es2015 and promises with bluebird were causing compilation problems. If we want to retain es5 as the compile target we'll have to address those issues.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -13,20 +13,20 @@
...
-    "tsd": "0.6.5",
-    "tslint": "3.11.0",
-    "typescript": "^2.1.4"
-  },
-  "scripts": {
-    "prepublish": "tsc --module umd --target es5 --moduleResolution node typings/tsd.d.ts index.ts bin/dts-generator.ts",
-    "test": "tests/run.sh"
+    "@types/glob": "^7.1.1",
+    "@types/mkdirp": "^0.5.2",
+    "dojo-loader": "^2.0.0-beta.7",
```
Comment (dylans) [inline package.json]: Do we still need the loader? I’m guessing we never released a non-beta version?

**Comment by dylans** [comment]
Need to determine why Travis is failing, but overall looks like some much needed changes. Thanks!

**Comment by dylans** [comment]
Looks like Travis for this may be running a very old version of Node.js still (4.1...)

File: package.json (hunk 1 above)
Comment (msssk) [reply package.json]: Build and tests run fine without `dojo-loader`; I removed it now.

**Pull request merged**